splunk-forwarder/main.tf

```diff
@@ -0,0 +1,111 @@
+resource "random_string" "unique_identfier" {
+  length  = 8
+  special = false
+}
+
+resource "aws_iam_role" "lambda_role" {
+  name = "lambda_role_${random_string.unique_identfier.result}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": "sdsd"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_cloudwatch" {
+  name        = "cw_for_lambda_${random_string.unique_identfier.result}"
+  description = "Lets the Lambda function write to cloudwatch"
```

Comment:
I'm confused, is this not supposed to write to S3? Does it combine the log streams first? Is the Java code open source also?

Reply:
In a perfect world I think it would make sense to have two policy statements then; one which allows the lambda to read from any log stream (or just the ones passed via vars), and one which allows it to create and write. Personally I've usually created the log group using terraform and given write privileges for that specific log group.

But I'm nitpicking here. Should not block a merge 😛